_posts/2019-01-23-multifile-python-functions.md

```diff
@@ -0,0 +1,213 @@
+
+    return process_tokens(words)
+
+def process_tokens(words: List[str]) -> Dict[str, int]:
+    """Normalize cases and remove plurals.
+    """
+    # ...
+
+def strip_punctuation(text: str) -> str:
+    # ...
+```
+
+Note that processing the `STOPWORDS` at the start of the file means it will be loaded into memory once the package is imported. In this case, once OpenFaaS deploys the function.
+
+Using an explicit relative import, we can very easily use the `process_text` method to create a very simple `handler.py`:
+
+```py
+# handler.py
+import json
+from .wordcloud import process_text
+
+
+def handle(req):
+    """handle a request to the function
+    Args:
+        req (str): request body
+    """
+
+    return json.dumps(process_text(req))
+```
+
+This style of relative imports will work for any file or sub-package you include inside of your function folder. Additionally, your IDE and linter will be able to resolve the imported code correctly!
+
+## Setup OpenFaaS
+
+OpenFaaS on Kubeneretes is my go-to deployment because I can use the built in k8s cluster in Docker-for-Mac.
+
+```sh
+$ kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+namespace "openfaas" created
+namespace "openfaas-fn" created
+
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/ && helm repo update
+"openfaas" has been added to your repositories
+Hang tight while we grab the latest from your chart repositories...
+...Successfully got an update from the "incubator" chart repository
+...Successfully got an update from the "contiamo" chart repository
+...Successfully got an update from the "openfaas" chart repository
+...Successfully got an update from the "stable" chart repository
+Update Complete. ⎈ Happy Helming!⎈
+
+$ helm upgrade openfaas --install openfaas/openfaas \
+    --namespace openfaas  \
+    --set basic_auth=false \
+    --set faasnetesd.imagePullPolicy='IfNotPresent' \
+    --set functionNamespace=openfaas-fn
+```
+
+This creates an OpenFaaS installation with auth disabled and it will use my locally built images for my functions.
+
+## Deploy and test the function
+
+```sh
+$ faas-cli build
+$ faas-cli deploy
+Deploying: wordcloud.
+
+Deployed. 202 Accepted.
+URL: http://127.0.0.1:31112/function/wordcloud
+
+$ curl -X POST http://127.0.0.1:31112/function/wordcloud \
+  -d 'This is some example text that we want to see a frequency response for.  It has text like apple, apples, apple tree, etc'
+{"example": 1, "text": 2, "want": 1, "see": 1, "frequency": 1, "response": 1, "for": 1, "apple": 3, "tree": 1, "etc": 1}
+```
+
+## Wrapping up
+
+Using relative imports allows the creation of Python functions that are split between several files. We could take this further and import from sub-folders or sub-folders of sub-folders even. This has the added benefit that the code is valid in both your local environment and the final docker container. Try the [completed code example in this repo.][project-repo]
```

Comment:
` We could take this further and import from sub-folders or sub-folders of sub-folders even`
->
` We could take this further and import from sub-folders or sub-folders of sub-folders`

Removes word `even` 

How about giving an example in the post?

Reply:
I am not sure about creating an real example that contains so many levels of nesting.  It would be a fairly complex example (which i don't have on hand) or very contrived.